provisioning/virtual-machines/ansible/roles/gather-desktop-environment-facts/tasks/main.yml

```
@@ -6,20 +6,30 @@
   register: startx_presence_check
   changed_when: False
   failed_when: False
+  when: desktop_mode is undefined
```

```suggestion
  when: install_desktop_environment is undefined or install_desktop_environment == "false"
```

Would be more explicit (same below)

---

Does not look resolved to me